### ISSUE TITLE
no-unreachable: Ignore type declarations 

### ISSUE THREAD

**Issue by littledivy** [denoland/deno_lint#619]
```typescript
throw new Error("ERROR");

type X = string;
```
```
(no-unreachable) This statement is unreachable
type X = string;
^^^^^^^^^^^^^^^^
```
Expected behaviour: The rule should ignore type decls.
